axelrod/fingerprint.py

```diff
@@ -0,0 +1,206 @@
+            else:
+                probe_dict[coord] = JossAnnTransformer((x, y))(probe)()
+        return probe_dict
+
+    def create_edges(self, probe_dict):
+        """Creates a set of edges for a spatial tournament.
+
+        Construncts edges that correspond to the probes in `probe_dict`. Probes
+        whose coordinates sum to less/more than 1 will have edges that link them
+        to 0/1 correspondingly.
+
+        Parameters
+        ----------
+        probe_dict : ordered dictionary
+            An Ordered Dictionary where the keys are tuples representing each
+            coordinate, eg. (x, y). The value is a `JossAnnTransformer` with
+            parameters that correspond to (x, y).
+
+        Returns
+        ----------
+        edges : list of tuples
+            A list containing tuples of length 2. All tuples will have either 0
+            or 1 as the first element. The second element is the index of the
+            corresponding probe (+2 to allow for including the Strategy and it's
+            Dual).
+        """
+        edges = []
+        for index, coord in enumerate(probe_dict.keys()):
+            #  Add 2 to the index because we will have to allow for the Strategy
+            #  and it's Dual
+            if sum(coord) > 1:
+                edge = (1, index + 2)
+            else:
+                edge = (0, index + 2)
+            edges.append(edge)
+        return edges
+
+    def fingerprint(self, turns=50, repetitions=10, granularity=0.01, cores=None):
+        """Build and play a spatial tournament.
+
+        Creates the probes and their edges then builds a spatial tournament
+        where the original strategy only plays probes whose coordinates sum to
+        less than 1 (or equal). Probes whose coordinates sum to more than 1 play
+        the Dual Strategy.
+
+        Parameters
+        ----------
+        turns : integer, optional
+            The number of turns per match
+        repetitions : integer, optional
+            The number of times the round robin should be repeated
+        granularity : float, optional
+            The seperation between each coordinate. Smaller granularity will
+            produce more coordinates that will be closer together.
+        cores : integer, optional
+            The number of processes to be used for parallel processing
+        """
+        self.probe_players = self.create_probes(self.probe, granularity)
+        self.edges = self.create_edges(self.probe_players)
+        original = self.strategy()
+        dual = DualTransformer()(self.strategy)()
+        probes = self.probe_players.values()
+        tourn_players = [original, dual] + list(probes)
+        spatial_tourn = axl.SpatialTournament(tourn_players, turns=turns,
+                                              repetitions=repetitions,
+                                              edges=self.edges)
+        print("Begin Spatial Tournament")
```

Comment:
No need for these print statements, pass a `progress_bar` argument to the `play` method (and to the `fingerprint` method).